File: c/739.c
int* dailyTemperatures(int* temp, int n, int* rs) {
    *rs = n;
    int ind = -1;
    int* s = (int*)malloc(sizeof(int) * n);
    int* res = (int*)malloc(sizeof(int) * n);
    for (int i = n - 1; i >= 0; --i) {
        while (ind >= 0 && temp[i] >= temp[s[ind]])
            --ind;
        res[i] = (ind < 0) ? 0 : (s[ind] - i);
        s[++ind] = i;
    }
    free(s);
    return res;
}

File: python/739.py
class Solution:
    def dailyTemperatures(self, temp: List[int]) -> List[int]:
        n = len(temp)
        ind = -1
        s = [0] * n
        res = [0] * n
        for i in range(n - 1, -1, -1):
            while ind >= 0 and temp[i] >= temp[s[ind]]:
                ind -= 1
            if ind < 0:
                res[i] = 0
            else:
                res[i] = s[ind] - i
            ind += 1
            s[ind] = i
        return res
